### ISSUE TITLE
Null Object Reference crash when Azra dies

### ISSUE THREAD

**Issue by IBwWG** [larsiusprime/tdrpg-bugs#719]
0.5.7win7x64

![dq-crash-azra-dies](https://cloud.githubusercontent.com/assets/16719964/14917217/3ed72bf0-0e1e-11e6-8bff-ed472d313dc8.png)

Different trace than #665:

```Called from ApplicationMain::main ApplicationMain.hx line 159
Called from ApplicationMain::create ApplicationMain.hx line 71
Called from lime.app.Application::exec lime/app/Application.hx line 222
Called from lime._backend.native.NativeApplication::exec lime/_backend/native/Na
tiveApplication.hx line 136
Called from lime._backend.native.NativeApplication::handleRenderEvent lime/_back
end/native/NativeApplication.hx line 368
Called from lime.app.Event_Void_Void::dispatch lime/app/Event.hx line 145
Called from *::_Function_2_1 lime/app/Application.hx line 110
Called from lime.app.Application::render lime/app/Application.hx line 762
Called from openfl.display.Stage::render openfl/display/Stage.hx line 677
Called from openfl.display.DisplayObjectContainer::__broadcast openfl/display/Di
splayObjectContainer.hx line 355
Called from openfl.display.DisplayObject::__broadcast openfl/display/DisplayObje
ct.hx line 251
Called from openfl.events.EventDispatcher::__dispatchEvent openfl/events/EventDi
spatcher.hx line 237
Called from flixel.FlxGame::onEnterFrame flixel/FlxGame.hx line 529
Called from flixel.FlxGame::step flixel/FlxGame.hx line 662
Called from com.leveluplabs.tdrpg.Main::update com/leveluplabs/tdrpg/Main.hx lin
e 1481
Called from flixel.FlxGame::update flixel/FlxGame.hx line 751
Called from flixel.FlxState::tryUpdate flixel/FlxState.hx line 168
Called from com.leveluplabs.tdrpg.State_Battle::update com/leveluplabs/tdrpg/Sta
te_Battle.hx line 880
Called from com.leveluplabs.tdrpg.State_Battle::battleUpdate com/leveluplabs/tdr
pg/State_Battle.hx line 995
Called from com.leveluplabs.tdrpg.BattleModel::update com/leveluplabs/tdrpg/Batt
leModel.hx line 1412
Called from com.leveluplabs.tdrpg.BattleModel::doUpdate com/leveluplabs/tdrpg/Ba
ttleModel.hx line 1816
Called from com.leveluplabs.tdrpg.BattleModel::updateEnemies com/leveluplabs/tdr
pg/BattleModel.hx line 3204
Called from com.leveluplabs.tdrpg.EnemySprite::updateFromModel com/leveluplabs/t
drpg/EnemySprite.hx line 510
Called from com.leveluplabs.tdrpg.EnemySprite::updateWalk com/leveluplabs/tdrpg/
EnemySprite.hx line 2155
Called from com.leveluplabs.tdrpg.EnemySprite::getNextWalk com/leveluplabs/tdrpg
/EnemySprite.hx line 1125
Called from flixel.util._FlxSignal.FlxSignal2::dispatch2 flixel/util/FlxSignal.h
x line 292
Called from com.leveluplabs.tdrpg.BattleModel::onEnemySignal com/leveluplabs/tdr
pg/BattleModel.hx line 2670
Called from com.leveluplabs.tdrpg.BattleModel::hitMcguffin com/leveluplabs/tdrpg
/BattleModel.hx line 2174
Called from com.leveluplabs.tdrpg.BattleModel::damageMcguffin com/leveluplabs/td
rpg/BattleModel.hx line 2202```

**Comment by IBwWG** [comment]
[export-crash-when-azra-dies.zip](https://github.com/larsiusprime/tdrpg-bugs/files/242515/export-crash-when-azra-dies.zip)

**Comment by larsiusprime** [comment]
I *believe* I've fixed this in 0.5.8

**Comment by IBwWG** [comment]
0.5.8 didn't post BTW (and 0.5.7 only posted windows still).
